src/transport/ExchangeMgr.h

```diff
@@ -0,0 +1,276 @@
+ *    over various transport mechanisms, for example, TCP, UDP, or CHIP Reliable Messaging.
+ *
+ */
+class DLL_EXPORT ExchangeContext
+{
+    friend class ExchangeManager;
+
+public:
+    typedef uint32_t Timeout; /**< Type used to express the timeout in this ExchangeContext, in milliseconds */
+
+    ExchangeManager * ExchangeMgr; /**< [READ ONLY] Owning exchange manager. */
+    uint64_t PeerNodeId;           /**< [READ ONLY] Node ID of peer node. */
+    uint16_t ExchangeId;           /**< [READ ONLY] Assigned exchange ID. */
+    void * AppState;               /**< Pointer to application-specific state object. */
+    bool AllowDuplicateMsgs;       /**< Boolean indicator of whether duplicate messages are allowed for a given exchange. */
+    uint32_t RetransInterval;      /**< Time between retransmissions (in milliseconds); 0 disables retransmissions. */
+    Timeout ResponseTimeout;       /**< Maximum time to wait for response (in milliseconds); 0 disables response timeout. */
+
+    enum
+    {
+        kSendFlag_AutoRetrans    = 0x0001, /**< Used to indicate that automatic retransmission is enabled. */
+        kSendFlag_ExpectResponse = 0x0002, /**< Used to indicate that a response is expected within a specified timeout. */
+        kSendFlag_DelaySend      = 0x0008, /**< Used to indicate that the sending of the current message needs to be delayed. */
+        kSendFlag_ReuseSourceId  = 0x0020, /**< Used to indicate that the source node ID in the message header can be reused. */
+        kSendFlag_RetainBuffer   = 0x0040, /**< Used to indicate that the message buffer should not be freed after sending. */
+        kSendFlag_AlreadyEncoded = 0x0080, /**< Used to indicate that the message is already encoded. */
+        kSendFlag_DefaultMulticastSourceAddress = 0x0100, /**< Used to indicate that default IPv6 source address selection should be
+                                                             used when sending IPv6 multicast messages. */
+        kSendFlag_FromInitiator    = 0x0200, /**< Used to indicate that the current message is the initiator of the exchange. */
+        kSendFlag_RequestAck       = 0x0400, /**< Used to send a RMP message requesting an acknowledgment. */
+        kSendFlag_NoAutoRequestAck = 0x0800, /**< Suppress the auto-request acknowledgment feature when sending a message. */
+
+        kSendFlag_MulticastFromLinkLocal = kSendFlag_DefaultMulticastSourceAddress,
+        /**< Deprecated alias for \c kSendFlag_DefaultMulticastSourceAddress */
+    };
+
+    bool IsInitiator() const;
+    bool IsResponseExpected() const;
+    void SetInitiator(bool inInitiator);
+    void SetResponseExpected(bool inResponseExpected);
+
+    CHIP_ERROR SendMessage(uint32_t protocolId, uint8_t msgType, System::PacketBuffer * msgPayload, uint16_t sendFlags = 0,
+                           void * msgCtxt = nullptr);
+    CHIP_ERROR SendCommonNullMessage();
```

Comment:
Need documentation for these functions..

Reply:
Added the document for all public APIs.

Reply:
Documentation is required here, but note: this is all private implementation inside of the Secure Channel box